File: easy-solution/galaxy4276/13.rs
impl Solution {
    pub fn roman_to_int(s: String) -> i32 {
        let mut result_number = 0;
        let mut last = 0;
        for c in s.chars() {
            let romanic_to_int = get_value(c);
            if last != 0 && last >= romanic_to_int {
                result_number += romanic_to_int;
            } else {
                let calculated = romanic_to_int - last - last;
                result_number += calculated;
            }
            last = romanic_to_int;
        }
        return result_number;
    }
}

fn get_value(c: char) -> i32 {
    match c {
        'I' => 1,
        'V' => 5,
        'X' => 10,
        'L' => 50,
        'C' => 100,
        'D' => 500,
        'M' => 1000,
        _ => 0,
    }
}

File: easy-solution/minesp3164/python/13.py
class Solution(object):
   def romanToInt(self, s):
      convert = {"I":1,"V":5,"X":10,"L":50,"C":100,"D":500,"M":1000,"IV":4,"IX":9,"XL":40,"XC":90,"CD":400,"CM":900}
      i = 0
      count = 0
      while i < len(s):
         if i + 1<len(s) and s[i :i + 2] in convert:
            count = count + convert[s[i: i + 2]]
            i = i + 2
         else:
            count = count + convert[s[i]]
            i = i + 1
      return count